cpp/tests/experimental/spatial/polygon_distance_test.cu

```diff
@@ -0,0 +1,482 @@
+            {0, 1, 2},
+            {0, 4, 8},
+            {{0.0, 0.0},
+             {2.0, 0.0},
+             {2.0, 2.0},
+             {0.0, 0.0},
+             {3.0, 3.0},
+             {3.0, 4.0},
+             {4.0, 4.0},
+             {3.0, 3.0}},
+            {0, 1, 2},
+            {0, 1, 2},
+            {0, 4, 8},
+            {{3.0, 0.0},
+             {4.0, 0.0},
+             {4.0, 1.0},
+             {3.0, 0.0},
+             {3.0, 3.0},
+             {3.0, 2.0},
+             {2.0, 2.0},
+             {3.0, 3.0}},
+            {1, 0});
+}
+
+TYPED_TEST(PairwisePolygonDistanceTest, TwoPairSinglePolygonHasHole)
+{
+  this->run({0, 1, 2},
+            {0, 1, 3},
+            {0, 4, 8, 12},
+            {{0.0, 0.0},
+             {2.0, 0.0},
+             {2.0, 2.0},
+             {0.0, 0.0},
+             {3.0, 3.0},
+             {3.0, 4.0},
+             {4.0, 4.0},
+             {3.0, 3.0},
+             {3.25, 3.5},
+             {3.5, 3.5},
+             {3.5, 3.75},
+             {3.25, 3.5}},
+            {0, 1, 2},
+            {0, 1, 2},
+            {0, 4, 8},
+            {{3.0, 0.0},
+             {4.0, 0.0},
+             {4.0, 1.0},
+             {3.0, 0.0},
+             {3.0, 3.0},
+             {3.0, 2.0},
+             {2.0, 2.0},
+             {3.0, 3.0}},
+            {1, 0});
+}
+
+TYPED_TEST(PairwisePolygonDistanceTest, TwoPairMultiPolygonNoHole)
+{
+  this->run({0, 1, 3},
+            {0, 1, 2, 3},
+            {0, 4, 8, 12},
+            {{0.0, 0.0},
+             {2.0, 0.0},
+             {2.0, 2.0},
+             {0.0, 0.0},
+             {3.0, 3.0},
+             {3.0, 4.0},
+             {4.0, 4.0},
+             {3.0, 3.0},
+             {3.0, 3.0},
+             {5.0, 3.0},
+             {4.0, 2.0},
+             {3.0, 3.0}},
+            {0, 1, 2},
+            {0, 1, 2},
+            {0, 4, 8},
+            {{3.0, 0.0},
+             {4.0, 0.0},
+             {4.0, 1.0},
+             {3.0, 0.0},
+             {3.0, 3.0},
+             {3.0, 2.0},
+             {2.0, 2.0},
+             {3.0, 3.0}},
+            {1, 0});
+}
+
+TYPED_TEST(PairwisePolygonDistanceTest, TwoPairMultiPolygonHasHole)
+{
+  this->run({0, 1, 3},
+            {0, 1, 2, 4},
+            {0, 4, 8, 12, 16},
+            {{0.0, 0.0},
+             {2.0, 0.0},
+             {2.0, 2.0},
+             {0.0, 0.0},
+
+             {3.0, 3.0},
+             {3.0, 4.0},
+             {4.0, 4.0},
+             {3.0, 3.0},
+
+             {3.0, 3.0},
+             {5.0, 3.0},
+             {4.0, 2.0},
+             {3.0, 3.0},
+
+             {3.5, 2.9},
+             {4.5, 2.9},
+             {4, 2.4},
+             {3.5, 2.9}},
+            {0, 1, 2},
+            {0, 1, 2},
+            {0, 4, 8},
+            {{3.0, 0.0},
+             {4.0, 0.0},
+             {4.0, 1.0},
+             {3.0, 0.0},
+             {3.0, 3.0},
+             {3.0, 2.0},
+             {2.0, 2.0},
+             {3.0, 3.0}},
+            {1, 0});
+}
```

Comment:
I have experimented to create some machinery to test the combinations more efficiently, instead of fully by manual here. But since I'm very down the road making these tests, I'll punt them to later.